src/Symfony/Component/Console/Application.php

```diff
@@ -128,13 +128,10 @@ public function run(InputInterface $input = null, OutputInterface $output = null
         }
 
         $renderException = function (\Throwable $e) use ($output) {
```

Comment:
We need to change

```
try {
            $exitCode = $this->doRun($input, $output);
        } catch (\Exception $e) {
```
to catch Throwable instead. Otherwise `$renderException` will never be called with a Throwable.
Do we only want to do this in SF 5 similar to HttpKernel? I don't really see why it could be a BC break to catch Throwable.

Reply:
You are right. The goal of this PR is to prepare for `catch (\Throwable)`. Personally I'm 👍 to catch them directly on 4.4 but it's technically a BC break IMHO. I think we need more opinions on this.

Reply:
> technically a BC break

What's the bc break?

Reply:
If a command throws an `\Error`:

Currently it doesn't go in this catch block, and it bubbles up directly. It always exits with the code 255 (exit code from the `register_shutdown_function` in `ErrorHandler::handleFatalError()`).

If we catch `\Throwable`, the exit code will be 1 by default or the one from error (cf the catch block).

So technically, it's a BC break, the exit code is not the same.